header-verifier/tests/header_verifier_blackbox_test.rs

```diff
@@ -1,125 +1,313 @@
+        operation_1,
+        Some(error_status),
     );
+}
+
+#[test]
+fn test_remove_executed_hash_no_esdt_address_registered() {
+    let mut state = HeaderVerifierTestState::new();
+    let bls_key_1 = ManagedBuffer::from("bls_key_1");
+    let managed_bls_keys = state.get_bls_keys(vec![bls_key_1]);
+
+    state.deploy_header_verifier_contract(managed_bls_keys);
+
+    let operation_1 = ManagedBuffer::from("operation_1");
+    let operation_2 = ManagedBuffer::from("operation_2");
+    let operation = state.get_mock_operation(vec![&operation_1, &operation_2]);
 
-    let first_operation: ManagedBuffer<StaticApi> =
-        ManagedBuffer::from("95cdb166d6e12a8c4a783a48d2e4f647e15fac4e5a115d4483f95881630a5433");
-    let second_operation =
-        ManagedBuffer::from("4851cd6e4a4799ad0d8e8ead37c88d930874302ab11edcc60f608654be14b2ed");
-
-    let mut bridge_operations: MultiValueEncoded<StaticApi, ManagedBuffer<StaticApi>> =
-        MultiValueEncoded::new();
-    bridge_operations.push(first_operation);
-    bridge_operations.push(second_operation);
-
-    state.world.sc_call(
-        ScCallStep::new()
-            .from(OWNER_ADDRESS_EXPR)
-            .call(state.header_verifier_contract.register_bridge_operations(
-                mock_signature,
-                bridge_operations_hash,
-                bridge_operations,
-            ))
-            .expect(TxExpect::user_error(
-                "str:Hash of all operations doesn't match the hash of transfer data",
-            )),
+    let error_status = ErrorStatus {
+        code: 4,
+        error_message: "There is no registered ESDT address",
+    };
+
+    state.propose_register_operations(operation.clone());
+    state.propose_remove_execute_hash(
+        &operation.bridge_operation_hash,
+        operation_1,
+        Some(error_status),
     );
 }
+
+#[test]
+fn test_remove_executed_hash() {
+    let mut state = HeaderVerifierTestState::new();
+    let bls_key_1 = ManagedBuffer::from("bls_key_1");
+    let managed_bls_keys = state.get_bls_keys(vec![bls_key_1]);
+
+    state.deploy_enshrine_esdt_contract(&managed_bls_keys);
+    state.deploy_header_verifier_contract(managed_bls_keys);
+
+    let operation_1 = ManagedBuffer::from("operation_1");
+    let operation_2 = ManagedBuffer::from("operation_2");
+    let operation = state.get_mock_operation(vec![&operation_1, &operation_2]);
+
+    state.propose_register_operations(operation.clone());
+    state.propose_register_esdt_address(ENSHRINE_ADDRESS);
+    state.propose_remove_execute_hash(&operation.bridge_operation_hash, operation_1, None);
+
+    state
+        .world
+        .query()
+        .to(HEADER_VERIFIER_ADDRESS)
+        .whitebox(header_verifier::contract_obj, |sc| {
+            let hash_of_hashes: ManagedBuffer<DebugApi> =
+                ManagedBuffer::from(operation.bridge_operation_hash.to_vec());
+            sc.pending_hashes(&hash_of_hashes).is_empty();
+        });
+}
```

Comment:
Also please make another test that will remove all the operations that were registered and expect:
- pending_hashes is empty
- hashes history still contains that hash_of_hashes (this should not be cleared)